scanner.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 // Trim off whitespace around the info - match least greedy, grab as much space on both sides
-var upgradeRegex = regexp.MustCompile(`UPGRADE "(.*)" NEEDED at height (\d+):\s+(.*?)\s*$`)
+var upgradeRegex = regexp.MustCompile(`UPGRADE "(.*)" NEEDED at height (\d+):\s+([^\s]*)\s*.*$`)
```

Comment:
If we greedy match until a space, I can simplify this further.
This also means the json must be compact with no spaces inside, that should be clearly documented, as the old code `{"binaries": {...}}` failed the test.

Simpler regexp:
`UPGRADE "(.*)" NEEDED at height (\d+):\s+([^\s]*)`
This greedy matches to the first space, and `.*$` is redundant